SdlStreaming/app/src/main/java/com/smartdevicelink/sdlstreaming/MainActivity.java

```diff
@@ -13,12 +21,49 @@
 
 public class MainActivity extends Activity {
 
+    private static final int BLUETOOTH_REQUEST_CODE = 200;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            if (!checkPermission()) {
+                requestPermission();
```

Comment:
```suggestion
                requestPermission();
                return;
```

Reply:
we don't want to call `SdlBroadcastReceiver.queryForConnectedService(this);` until they have accepted the permission if they are using an Android 12 device.